Kubernetes documentation enhancements (#4374)
woodpecker-ci/woodpecker: docs/docs/30-administration/22-backends/40-kubernetes.md

`serviceAccountName` specifies the name of the ServiceAccount which the Pod will mount. This service account must be created externally.
See the [Kubernetes documentation](https://kubernetes.io/docs/concepts/security/service-accounts/) for more information on using service accounts.

```yaml
steps:
  - name: 'My kubernetes step'
    image: alpine
    commands:
      - echo "Hello world"
    backend_options:
      kubernetes:

>         # Use the service account `default` in the current current namespace.

Reviewer: ```suggestion
        # Use the service account `default` in the current namespace.
```

        # This usually the same as wherever woodpecker is deployed.
        serviceAccountName: default
```

> To give steps access to the kubernetes api via service account, look at [RBAC Authorization](https://kubernetes.io/docs/reference/access-authn-authz/rbac/)

Reviewer 2: ```suggestion
To give steps access to the Kubernetes API via service account, take a look at [RBAC Authorization](https://kubernetes.io/docs/reference/access-authn-authz/rbac/)
```

### Volumes

To mount volumes a PersistentVolume (PV) and PersistentVolumeClaim (PVC) are needed on the cluster which can be referenced in steps via the `volumes` option.

Persistent volumes need to be manually generated, use the Kubernetes [Persistent Volumes](https://kubernetes.io/docs/concepts/storage/persistent-volumes/) documentation as a reference.

_If your PVC is not highly available or NFS based, you may need to also integrate affinity settings to ensure your steps run on the correct node._

> NOTE: If you plan to use this volume on more than one workflow concurrently, make sure you have configured the PVC with:

Reviewer 2: ```suggestion
NOTE: If you plan to use this volume in more than one workflow concurrently, make sure you have configured the PVC in `RWX` mode. Keep in mind that this feature must be supported by the used CSI driver:
```

    [...]
```

Or as follows when using a normal image:

```yaml
steps:
  - name: "Edit cache"
    image: alpine:latest
    volumes:
      - woodpecker-cache:/woodpecker/src/cache
    commands:
      - echo "Hello World" > /woodpecker/src/cache/output.txt
    [...]
```

> Please note that `settings:` is reserved for use with plugins, and is incompatible with `commands:`

Reviewer: This is unnecessary here.

Reviewer: https://woodpecker-ci.org/docs/usage/workflow-syntax#commands

> Default: empty

Secret names to pull images from private repositories. See, how to [Pull an Image from a Private Registry](https://kubernetes.io/docs/tasks/configure-pod-container/pull-image-private-registry/).

## Metrics

> 

Reviewer: And this. Move in deployment instruction or in helm chart repository.
https://woodpecker-ci.org/docs/administration/deployment-methods/kubernetes
